packages/tailwindcss/src/utilities.ts

```diff
@@ -301,6 +302,12 @@ export function createUtilities(theme: Theme) {
 
         // If there is still no value but the utility supports bare values, then
         // use the bare candidate value as the value.
+        if (value === null && candidate.negative && desc.handleNegativeBareValue) {
+          value = desc.handleNegativeBareValue(candidate.value)
+          if (!value?.includes('/') && candidate.modifier) return
+          if (value !== null) return desc.handle(value)
+        }
+
```

Comment:
This isn't strictly necessary for things to just _work_ but without control over this from `spacingUtility`, the output of negative values would contain nested `calc` calls and just looked really ugly and unnecessary:

```diff
  .-ml-4 {
-   margin-left: calc(var(--spacing) * -4);
+   margin-left: calc(calc(var(--spacing) * 4) * -1);
  }
```